tests/aws/services/s3/test_s3.py

```diff
@@ -4251,16 +4257,21 @@ def test_create_bucket_head_bucket(self, snapshot, aws_client):
                 snapshot.transform.regex(r"s3\.amazonaws\.com", "<host>"),
                 snapshot.transform.regex(r"s3\.localhost\.localstack\.cloud:4566", "<host>"),
                 snapshot.transform.regex(r"s3\.localhost\.localstack\.cloud:443", "<host>"),
+                snapshot.transform.key_value("Location", reference_replacement=False),
```

Comment:
This was a very specific snapshotting to check the format of the response, and now it's totally overridden by this new line. This test is testing the behaviour of `us-east-1` vs all other regions, so it should stay that way.

We need to have it hardcoded to `us-east-1` for the first bucket, and whatever region for the second, but it is really important.

The snapshot would fail anyway when run into multi-account because of the `if` check and the snapshot hard matching against `us-east-2`.